galera-loader

```diff
@@ -28,19 +28,40 @@ backup_wait(){
 }
 
 backup(){
+	BACKUP_LOCK=/var/tmp/backup_lock
+	if [ -z "$BACKUP_COUNT" ]; then
+		BACKUP_COUNT=0
+	fi
+	if [ -f "$BACKUP_LOCK" ]; then
+		log "0 [Warning] Backup: Skipping due to lock file existing"
+		return 1
+	fi
+	touch $BACKUP_LOCK
 	log "0 [Note] Backup: Starting"
 	cd /var/backups/mysql
 	# backup everything to this directory
-	for DB in $(
-	mysql -Ns -e 'show databases' \
-	| grep -vE "^information_schema$|^performance_schema$|^test$"
-	); do
-	log "0 [Note] Backup: Backing up $DB"
-	/usr/bin/mysqldump --single-transaction --databases "$DB" \
-	> "${DB}.sql"
-	log "0 [Note] Backup: Backing up $DB done"
+	for DB in $( mysql -Ns -e 'show databases' \
+				| grep -vE "^information_schema$|^performance_schema$|^test$"
+				); do
+		if [ "$BACKUP_COUNT" != "0" ]; then
+			for count_up in $(seq 1 $BACKUP_COUNT); do
+				let count_down=$BACKUP_COUNT-$count_up+1
+				let count_previous=$count_down-1
```

Comment:
This is the last statement exposed by `set -x`. I don't know why it's dying here.